Tests/FluentAssertions.Specs/Specialized/TaskOfTAssertionSpecs.cs

```diff
@@ -54,6 +55,53 @@ public async Task When_task_completes_fast_async_it_should_succeed()
             await action.Should().NotThrowAsync();
         }
 
+        [Fact]
+        public async Task When_task_completes_async_and_result_is_not_expected_it_should_throw()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<int>();
+
+            // Act
+            Func<Task> action = async () =>
+            {
+                Func<Task<int>> funcSubject = () => taskFactory.Task;
+
+                (await funcSubject.Should(timer).CompleteWithinAsync(100.Milliseconds()))
+                    .Which.Should().Be(42);
+            };
+
+            taskFactory.SetResult(99);
+            timer.Complete();
+
+            // Assert
+            // TODO message currently shows "Expected (await funcSubject to be...", but should be "Expected funcSubject to be...",
```

Comment:
I haven't used it myself yet, but I think decorating `WithResult` with [`[CustomAssertion]`](https://fluentassertions.com/introduction#subject-identification) can solve this.

Reply:
The `WithResult` extension is not affected.
The previous one is affected because of `(await foo).Which.Should().Be(...)`.